pkg/validation/spec/gnostic_test.go

```diff
@@ -0,0 +1,507 @@
+				c.Fuzz(&v.Description)
+				c.Fuzz(&v.Required)
+
+				v.Schema = &Schema{}
+				c.Fuzz(&v.Schema)
+
+			} else {
+				c.Fuzz(&v.SimpleSchema)
+				c.Fuzz(&v.CommonValidations)
+				v.AllowEmptyValue = false
+				v.Description = c.RandString() + "x"
+				v.Name = c.RandString() + "x"
+
+				switch c.Intn(4) {
+				case 0:
+					// Header param
+					v.In = "header"
+				case 1:
+					// Form data param
+					v.In = "formData"
+					v.AllowEmptyValue = c.RandBool()
+				case 2:
+					// Query param
+					v.In = "query"
+				case 3:
+					// Path param
+					v.In = "path"
+					v.Required = true
+				default:
+					panic("unreachable")
+				}
+			}
+		},
+		func (v *Schema, c fuzz.Continue) {
+			c.Fuzz(&v.SchemaProps)
+			c.Fuzz(&v.SwaggerSchemaProps)
+			c.Fuzz(&v.VendorExtensible)
+			// c.Fuzz(&v.ExtraProps)
+
+			// Not supported by official openapi v2 spec
+			// and stripped by k8s apiserver
+			v.ID = ""
+			v.AnyOf = nil
+			v.OneOf = nil
+			v.Not = nil
+			v.Nullable = false
+			v.AdditionalItems = nil
+			v.Schema = ""
+			v.PatternProperties = nil
+			v.Definitions = nil
+			v.Dependencies = nil
+		},
+	)
+
+	expected := Swagger{}
+	fuzzer.Fuzz(&expected)
+
+	// Convert to gnostic via JSON to compare
+	jsonBytes, err := json.Marshal(expected)
+	require.NoError(t, err)
+
+	t.Log("Specimen", string(jsonBytes))
+
+	gnosticSpec, err := openapi_v2.ParseDocument(jsonBytes)
+	assert.NoError(t, err)
+
+	actual := Swagger{}
+	require.NoError(t, actual.FromGnostic(gnosticSpec))
+	require.EqualValues(t, expected, actual)
+
+	newJsonBytes, err := json.Marshal(actual)
+	require.NoError(t, err)
+	require.Equal(t, string(jsonBytes), string(newJsonBytes))
+}
+
+func TestGnosticConversionSmallDeterministic(t *testing.T) {
+	gnosticCommonTest(
+		t,
+		fuzz.
+			NewWithSeed(15).
+			NilChance(0.8).
+			MaxDepth(10).
+			NumElements(1, 2),
+	)
+}
+
+func TestGnosticConversionSmallDeterministic2(t *testing.T) {
+	// A failed case of TestGnosticConversionSmallRandom
+	// which failed during development/testing loop
+	gnosticCommonTest(
+		t,
+		fuzz.
+			NewWithSeed(1646770841).
+			NilChance(0.8).
+			MaxDepth(10).
+			NumElements(1, 2),
+	)
+}
+
+func TestGnosticConversionSmallDeterministic3(t *testing.T) {
+	// A failed case of TestGnosticConversionSmallRandom
+	// which failed during development/testing loop
+	gnosticCommonTest(
+		t,
+		fuzz.
+			NewWithSeed(1646772024).
+			NilChance(0.8).
+			MaxDepth(10).
+			NumElements(1, 2),
+	)
+}
+
+func TestGnosticConversionSmallDeterministic4(t *testing.T) {
+	// A failed case of TestGnosticConversionSmallRandom
+	// which failed during development/testing loop
+	gnosticCommonTest(
+		t,
+		fuzz.
+			NewWithSeed(1646791953).
+			NilChance(0.8).
+			MaxDepth(10).
+			NumElements(1, 2),
+	)
+}
+
+
+func TestGnosticConversionSmallRandom(t *testing.T) {
+	seed := time.Now().Unix()
+	t.Log("Using seed: ", seed)
+
+	gnosticCommonTest(
+		t,
+		fuzz.
+			NewWithSeed(seed).
+			NilChance(0.8).
+			MaxDepth(10).
```

Comment:
I disagree that the "deterministic" tests add no value. The `SmallRandom` test case does as you described to fuzz for new failing test cases. Once a failing case is discovered by the random fuzzer it is easy to take the seed used to reproduce the failure, fix it, and add a regression test. Do this enough times and you have a corpus of tricky test cases the code is known to have failed in the past against to guard against regression.

To fuzz in a loop 100 times without printing the random seed used for each case offers no value to a developer other than to say "it's broken", since the exact case can't be reproduced.

Reply:
> even changing parameters like this. I don't think the "size" really has an impact for these.

I was forced to change the parameters since the fuzz to generate the object would take too long otherwise. The different "sizes" was useful for me during development to work on smaller cases and then larger and larger. The distinction may not be useful as part of the test sweet but I see no need to remove them.